app/controllers/application_controller.rb

```diff
@@ -1,12 +1,12 @@
 class ApplicationController < ActionController::Base
+  include Pagy::Backend 
   helper_method :current_user
 
+  def authenticate_user!
+    redirect_to request.referer || root_path unless current_user
+  end
+
   def current_user
-    if session[:user_id]
-      @cureent_user ||= User.find(session[:user_id])
-    else
-      @cureent_user = nil
-    end
+    @current_user ||= User.find_by(id: session[:user_id]).presence
```

Comment:
no need for `.presence` `find_by` will return `nil`

Reply:
done